### PR TITLE
Adds logic to calculate run duration based on previous run pace

### PR DESCRIPTION
**Pull request by markstanden**
## Summary by CodeRabbit

- New Features
  - Personalised, effort‑aware run recommendations with duration tuned to recent history, explicit rest‑day support, and a backend API to request recommendations; new default pacing and rest semantics.

- Bug Fixes
  - Negative distances clamped/validated; zero‑distance (rest) handled gracefully; clearer validation and error messaging for invalid distances.

- Refactor
  - Recommendation flow split into distance, effort and duration steps with effort‑level pace fallback.

- Tests
  - Expanded unit tests and test helpers covering pacing, effort fallback, rest days and edge cases.

### CODE REVIEW

File: src/Shared/Constants/Runs.cs [hunk 1]
```diff
@@ -14,20 +17,20 @@ public static class InsufficientHistory
...
         public const int RunDistanceMetres = 1000;
         public const byte RunEffort = EffortLevel.Easy;
-        public const int RunDurationMinutes = 15;
+        public const int RunPaceMinsPerKm = 15;
 
-        public static readonly TimeSpan RunDurationTimeSpan
-            = TimeSpan.FromMinutes(RunDurationMinutes);
+        public static TimeSpan RunDurationTimeSpan(int distanceMetres)
+            => TimeSpan.FromMinutes(distanceMetres / 1000D * RunPaceMinsPerKm);
     }
```
[the remark this answers is not in the document]
Comment (markstanden) [reply src/Shared/Constants/Runs.cs]: @coderabbitai - yes additional validation would help here - I've also added tests for it, and added logic for handling zero distances inline with rest day logic
